webapp-django/crashstats/symbols/tests/test_views.py

```diff
@@ -298,23 +298,45 @@ def test_upload(self):
         )
 
         url = reverse('symbols:upload')
-        response = self.client.get(url)
-        eq_(response.status_code, 405)
 
         with self.settings(MEDIA_ROOT=self.tmp_dir):
             with open(ZIP_FILE, 'rb') as file_object:
                 response = self.client.post(
                     url,
                     {'file.zip': file_object},
+                    # note! No HTTP_AUTH_TOKEN
+                )
+                eq_(response.status_code, 403)
+
+            with open(ZIP_FILE, 'rb') as file_object:
+                response = self.client.post(
+                    url,
+                    {'file.zip': file_object},
+                    HTTP_AUTH_TOKEN=''
+                )
+                eq_(response.status_code, 403)
+
+            with open(ZIP_FILE, 'rb') as file_object:
+                response = self.client.post(
+                    url,
+                    {'file.zip': file_object},
+                    HTTP_AUTH_TOKEN='somejunk'
+                )
+                eq_(response.status_code, 403)
+
+            with open(ZIP_FILE, 'rb') as file_object:
+                response = self.client.post(
+                    url,
+                    {'file.zip': file_object},
                     HTTP_AUTH_TOKEN=token.key
                 )
-            eq_(response.status_code, 201)
-            symbol_upload = models.SymbolsUpload.objects.get(user=user)
-            eq_(symbol_upload.filename, 'file.zip')
-            ok_(symbol_upload.size)
-            ok_(symbol_upload.file)
-            ok_(symbol_upload.file_exists)
-            ok_(symbol_upload.content)
+                eq_(response.status_code, 201)
+                symbol_upload = models.SymbolsUpload.objects.get(user=user)
+                eq_(symbol_upload.filename, 'file.zip')
+                ok_(symbol_upload.size)
+                ok_(symbol_upload.file)
+                ok_(symbol_upload.file_exists)
+                ok_(symbol_upload.content)
```

Comment:
Stylistic. It's because these lines belong to the last attempt of opening a ZIP_FILE.